src/crc_prediction_markets/BetYes.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./IFixedProductMarketMaker.sol";
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import "circles-v2/hub/Hub.sol";
+import "circles-v2/lift/IERC20Lift.sol";
+
+import {console} from "forge-std/console.sol";
+
+contract BetYesContract is ERC1155Holder {
+    IFixedProductMarketMaker public immutable fpmm;
+    //ERC20 public immutable groupCRCToken;
+    uint256 public immutable outcomeIndex;
+    Hub public hub;
+    address groupCRCToken;
+
+    //ToDo - balance per user for later redeem
+
+    constructor(address fpmmAddress, address _groupCRCToken, uint256 _outcomeIndex, address _hubAddress) {
+        fpmm = IFixedProductMarketMaker(fpmmAddress);
+        groupCRCToken = _groupCRCToken;
+        outcomeIndex = _outcomeIndex;
+        hub = Hub(_hubAddress);
+    }
+
+    function placeBet(uint256 investmentAmount) public {
+        // Wrap received tokens
+        console.log("entered placebet, groupCRCToken", address(groupCRCToken));
+        // not needed
+        address erc20Group = hub.wrap(address(groupCRCToken), investmentAmount, CirclesType.Inflation);
```

Comment:
@Max
after wrapping the investment amount the amount of the inflationary tokens will be greater than the investmentAmount
because demurrage is not applied to the inflationary tokens amount itself

Reply:
fetching balance before/after, hence closing this